.github/workflows/scripts/check-if-issue-and-pr-exist.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# The API URLs in conditions below search for any existing pull requests and issues using wildcard and
+# generates ISSUE_EXISTS and PR_EXISTS environment variables and places those in the environment for next steps
+# through $GITHUB_ENV;
+# See https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-environment-variable
+# See docs: https://docs.github.com/en/rest/reference/search#search-issues-and-pull-requests--parameters
+# If the issue exists we also set the `ISSUE_NUMBER` environment variable.
+
+PR_NUMBER=$(curl -s --request GET \
+  --url https://api.github.com/search/issues\?q=is:pull-request+is:open+repo:$REPO+in:title+Submodule-sync+$MATRIX_PATH+Update+submodule+to+its+latest+version \
```

Comment:
The contents of this line make this script not as generic as its name and comments would suggest. It doesn't check for _any_ issue and/or PR to exist, it checks for the _submodule sync_ issue and/or PR.

We can either:
A) Name this file something more specific (e.g. `scripts/submodule-sync/check-if-issue-and-pr-exist`)
B) Make this file more generally useful (i.e. take the name of the pull request and issue as parameters).

Reply:
I have added B - to pass PR and Issue titles as parameters.